### ISSUE TITLE
Player crouched state resets after setPedAnimation() was called

### ISSUE THREAD

**Issue by Einheit-101** [multitheftauto/mtasa-blue#462]
**Describe the bug**
As soon as you use `setPedAnimation()` while the player is crouched, he will stand up after finishing the animation. This bug now hits especially hard because custom animations open a new world for scripters.

**To Reproduce**
1. Go to map editor, press F5 for testing
2. Crouch the player
3. Open freeroam menu with F1, open "anim" tab
4. Apply any animation and stop it again
5. Player will stand up after that

**Expected behavior**
The player should stay crouched, this bug makes it hard / impossible to play certain animations while crouching (For example "crouch_roll_r" will let the player roll right but he stands up, which is not what i want)

**MTA Version 1.5.6-release-14378**

**Comment by patrikjuvonen** [comment]
#488 might be related

**Comment by ghost** [comment]
@qaisjp What do you think? When setPedAnimation is done executing, the crouch state should be restored if ped was ducked?
@Einheit-101 Does this happen with other tasks as well? or is it only crouch?

**Comment by Einheit-101** [comment]
Uhm i dont know of any task that could get cut off, i think you stop climbing but thats expected behaviour.

**Comment by qaisjp** [comment]
> @qaisjp What do you think? When setPedAnimation is done executing, the crouch state should be restored if ped was ducked?

What do you think about an extra argument, `retainPedState`?  The wording is vague enough for other similar things to be clustered into this setting later.

**Comment by Einheit-101** [comment]
This sounds good, but even better may be a "pedState" setting: "crouching" or "standing", the Ped/Player will do whatever you write in that argument when the Animation is finished.

**Comment by ghost** [comment]
> 
> What do you think about an extra argument, `retainPedState`? The wording is vague enough for other similar things to be clustered into this setting later.

I like @Einheit-101 's idea. However, I'd prefer to pass the name of a control state which will be set after the ped is done playing the animation. I'll do it if it's not so difficult to accomplish.

**Comment by ghost** [comment]
Fixed in 49ffc913457b9431f23e62b28d7fad6eff41b695 . Let me know if there's any problem.

**Comment by Einheit-101** [comment]
Thanks a lot, i will check as soon as possible!

**Comment by qaisjp** [comment]
Didn't get time to respond earlier. That idea works too, @Einheit-101!

The reason I suggested just a simple `retainState` is so that we can be forwards compatible and reuse the arg for any other similar problem in the future (instead of repeatedly adding new args) -- we would advertise the fact that this arg may "retain more state" in the future.

So instead of specifying "crouching", you would just use regular Lua functions to go do that ( `setPedDucked` #488 or any other setPedControlState ), and then use `retainState=true` to prevent `setPedAnimation` from undoing those changes.

I think currently your commit (49ffc91) is basically the same as what I'm suggesting, since your code will force the arg `false` if the ped is not currently ducked.

tl;dr I'm suggesting the arg is renamed (in just the wiki / Lua interface) so that we can reuse the arg for future states.

What are both your thoughts on that or do you think there won't be any other attributes that need to be retained?

**Comment by ghost** [comment]
> 
> tl;dr I'm suggesting the arg is renamed (in just the wiki / Lua interface) so that we can reuse the arg for future states.
> 
> What are both your thoughts on that or do you think there won't be any other attributes that need to be retained?

Okay, that sounds good.

**Comment by Einheit-101** [comment]
Can the crouch state be reset a bit "earlier"?
Because the player always attempts to stand up after an animation is finished, he then quickly crouches again, but it is not really smooth.
It can be seen for example here:
crun setPedAnimation(localPlayer, "ped", "Crouch_Roll_L", -1, false, true, false, false, 0, true)

The difference is not huge with this one, but it is very good visible when using my prone animation.

**Comment by ghost** [comment]
> 
> Can the crouch state be reset a bit "earlier"?

I think that's hardcoded in the crouch task as it's responsible for playing the crouch animation. It's better to create a separate issue for this.
